### PR TITLE
Display configuration summary and number of issues

### LINKED ISSUE #8045: Display configuration summary and number of issues

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: projects/assets/src/translations/en/cart.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ export const cart = {
     total: 'Total',
     cartTotal: 'Cart total ({{count}} item)',
     cartTotal_plural: 'Cart total ({{count}} items)',
+    numberOfIssue: '{{errors}} error must be resolved before checkout.',
```
Comment (ChristophHi) [inline projects/assets/src/translations/en/cart.ts]: For the singular version no parameter is needed

File: projects/core/src/occ/occ-models/occ.models.ts [hunk 2]
```diff
@@ -1431,6 +1431,52 @@ export namespace Occ {
...
+    statusSummaryList?: StatusSummary[];
+    /**
+     * @member {ConfigurationInfo[]} [configurationInfos]
+     */
+    configurationInfos?: ConfigurationInfo[];
+  }
+
+  /**
+   *
+   * An interface representing ConfigurationInfo.
```
Comment (ChristophHi) [inline projects/core/src/occ/occ-models/occ.models.ts:1446]: Please describe in a more detailed way what this is about. Important to mention that this is not bound to a specific configurator but a data that is common for all configurators.

File: projects/core/src/occ/occ-models/occ.models.ts [hunk 3]
```diff
@@ -1431,6 +1431,52 @@ export namespace Occ {
...
+    configuratorType?: string;
+    /**
+     * @member {string} [status]
+     */
+    status?: string;
+  }
+
+  /**
+   *
+   * An interface representing StatusSummary.
```
Comment (ChristophHi) [inline projects/core/src/occ/occ-models/occ.models.ts:1470]: Also please describe in more detail

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html [hunk 4]
```diff
@@ -1,170 +1,219 @@
...
-            [routerLink]="{ cxRoute: 'product', params: item.product } | cxUrl"
-            (click)="viewItem()"
-            >{{ item.product.name }}</a
-          >
-        </div>
-        <div *ngIf="item.product.code" class="cx-code">
-          {{ 'cartItems.id' | cxTranslate }} {{ item.product.code }}
-        </div>
-        <!-- Generic Configuration-->
+<ng-container *ngIf="item.statusSummaryList[0]?.numberOfIssues > 0">
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: This does not work for non-configurable items or items without status, cart display breaks
(Cannot read property '0' of undefined). We need to check for item.statusSummaryList !== undefined as well and should introduce a component function for that to not code this in the view

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html [hunk 5]
```diff
@@ -1,170 +1,219 @@
...
+  <div [ngClass]="compact ? 'cx-compact row' : 'row'">
+    <div class="cx-error-container">
+      <div [ngClass]="compact ? 'col-2' : 'cx-error-icon-container'">
+        <cx-icon class="cx-error-icon" [type]="iconTypes.ERROR"></cx-icon>
+      </div>
+      <div [ngClass]="compact ? 'col-10' : 'cx-error-msg-container'">
+        {{
+          getIssueMessageKey(item.statusSummaryList[0]?.numberOfIssues)
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: No, we need to take the status of the summary list into account! For the textfield configurator, you will have 3 success statuses (which are displayed as errors currently). Probably we should get the issues also from the component

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts [hunk 6]
```diff
@@ -205,4 +206,93 @@ describe('CartItemComponent', () => {
       );
     });
   });
+
+  describe('Depicting configurable products in the cart', () => {
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts:210]: Add tests for status summary containing successes (see above)

File: projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts [hunk 7]
```diff
@@ -40,5 +43,19 @@ export class ConfigureCartEntryComponent {
     return this.readOnly;
   }
 
-  constructor(private genericConfigUtilsService: GenericConfigUtilsService) {}
+  closeActiveModal(): void {
+    const translation = this.getDisplayOnly()
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts]: Please specify type

File: projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts [hunk 8]
```diff
@@ -40,5 +43,19 @@ export class ConfigureCartEntryComponent {
...
+      ? this.translation.translate('configurator.header.displayConfiguration')
+      : this.translation.translate('configurator.header.editConfiguration');
+
+    translation
+      .pipe(take(1))
+      .subscribe((reason) => this.modalService.closeActiveModal(reason));
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts]: Just a question: Why do we need to provide this reason?

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: projects/core/src/occ/occ-models/occ.models.ts (hunk 2 above)
Comment (steinsebastian) [reply projects/core/src/occ/occ-models/occ.models.ts:1446]: I'll do. But all other interfaces in that file (except enum definitions) are documented in the same way.

File: projects/core/src/occ/occ-models/occ.models.ts (hunk 3 above)
Comment (steinsebastian) [reply projects/core/src/occ/occ-models/occ.models.ts:1470]: done.

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html (hunk 4 above)
Comment (Larisa-Staroverova) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: We can add "**?**" after item like <ng-container *ngIf="item**?**.statusSummaryList[0]?.numberOfIssues > 0">

Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: This is not sufficient. I'm working on this.

File: projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts (hunk 8 above)
Comment (ChristophHi) [reply projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts]: As per our conversation Larisa: Please remove localization here

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html (hunk 4 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: Done.

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html (hunk 5 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: Will do

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts (hunk 6 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts:210]: will do

File: projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts (hunk 8 above)
Comment (Larisa-Staroverova) [reply projects/storefrontlib/src/cms-components/configurator/generic/configure-cart-entry/configure-cart-entry.component.ts]: Done

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts (hunk 6 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.spec.ts:210]: done

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html (hunk 5 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: done

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts [hunk 9]
```diff
@@ -58,6 +62,22 @@ export class CartItemComponent implements OnInit {
     );
   }
 
+  hasIssues(): boolean {
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts:79]: Please add JSDoc

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts [hunk 10]
```diff
@@ -58,6 +62,22 @@ export class CartItemComponent implements OnInit {
...
+    return this.getNumberOfIssues() > 0;
+  }
+
+  getNumberOfIssues(): number {
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts:88]: Also please add JSDoc

File: projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html [hunk 11]
```diff
@@ -1,170 +1,218 @@
...
+          </div>
+          <div *ngIf="item.product.code" class="cx-code">
+            {{ 'cartItems.id' | cxTranslate }} {{ item.product.code }}
+          </div>
           <div
-            *ngFor="
-              let variant of item.product.baseOptions[0]?.selected
-                ?.variantOptionQualifiers
+            class="cx-configuration-info-container"
+            *ngIf="
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.html]: This check we should also have in the component

**Review by ChristophHi — CHANGES_REQUESTED** [review]

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**